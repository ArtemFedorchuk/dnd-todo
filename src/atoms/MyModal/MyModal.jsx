import { Modal } from '@mui/base/Modal';
import clsx from 'clsx';
import PropTypes from 'prop-types';
import React, { forwardRef } from 'react';

import styles from './styles.module.scss';

import { MyButton } from '../MyButton/MyButton';

const Backdrop = forwardRef((props, ref) => {
  const { open, className, ...other } = props;
  return <div className={clsx({ 'MuiBackdrop-open': open }, className)} ref={ref} {...other} />;
});

Backdrop.displayName = 'Backdrop';

Backdrop.propTypes = {
  className: PropTypes.string.isRequired,
  open: PropTypes.bool,
};

export const MyModal = ({ children, open, onClose, onConfirm, title }) => {
  return (
    <>
      {open && (
        <Backdrop className={styles.backdrop}>
          <Modal
            id={'Modal'}
            className={styles.modal}
            aria-labelledby="unstyled-modal-title"
            aria-describedby="unstyled-modal-description"
            open={open}
            onClose={onClose}
          >
            <div className={styles.modalContentWrapper}>
              <div>
                <div className={styles.modalHeader}>
                  <h3>{title}</h3>
                  <MyButton variant="secondary" text="x" onClick={onClose} />
                </div>
                {children}
              </div>
              <div className={styles.modalFooter}>
                <MyButton variant="secondary" text="Cancel" onClick={onClose} />
                <MyButton
                  variant="error"
                  text="Confirm"
                  onClick={() => {
                    onConfirm?.();
                    onClose?.();
                  }}
                />
              </div>
            </div>
          </Modal>
        </Backdrop>
      )}
    </>
  );
};

MyModal.propTypes = {
  children: PropTypes.oneOfType([PropTypes.arrayOf(PropTypes.node), PropTypes.node]).isRequired,
  open: PropTypes.bool.isRequired,
  onClose: PropTypes.func,
  onConfirm: PropTypes.func,
  title: PropTypes.string.isRequired,
};
