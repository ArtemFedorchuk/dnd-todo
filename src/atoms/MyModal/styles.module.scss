@import '../../styles/main';

.backdrop {
  position: fixed;
  z-index: 1;
  inset: 0;

  background-color: $backdrop-bg;

  -webkit-tap-highlight-color: transparent;
}

.modal {
  position: fixed;
  z-index: 1300;
  inset: 0;

  display: flex;
  align-items: center;
  justify-content: center;
}

.modalContentWrapper {
  display: flex;
  flex-direction: column;
  justify-content: space-between;

  width: 400px;
  height: 300px;
  padding: 20px;

  background: $white;
  border-radius: 5px;
}

.modalHeader {
  display: flex;
  justify-content: space-between;
  padding-bottom: 20px;
}

.modalFooter {
  display: flex;
  justify-content: space-between;
}
