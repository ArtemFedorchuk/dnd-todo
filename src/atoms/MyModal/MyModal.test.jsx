import { render, screen } from '@testing-library/react';

import '@testing-library/jest-dom/extend-expect';
import { MyModal } from './MyModal';

describe('Modal component', () => {
  it('List modals', () => {
    render(
      <MyModal open={true} title="Modal title">
        Some content
      </MyModal>,
    );

    expect(screen.queryByText('Some content')).toBeInTheDocument();
  });

  it('Modal snapshot', () => {
    const modalElem = render(
      <MyModal open={false} title="Modal title">
        Some content
      </MyModal>,
    );

    expect(modalElem).toMatchSnapshot();
  });
});
