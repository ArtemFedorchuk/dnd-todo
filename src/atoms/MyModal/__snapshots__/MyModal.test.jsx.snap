// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MyModal modal snapshot 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body
    style="padding-right: 1024px; overflow: hidden;"
  >
    <div
      aria-hidden="true"
    >
      <div
        class="backdrop"
      />
    </div>
    <div
      aria-describedby="unstyled-modal-description"
      aria-labelledby="unstyled-modal-title"
      class="MuiModal-root modal"
      id="Modal"
      role="presentation"
    >
      <div
        data-testid="sentinelStart"
        tabindex="0"
      />
      <div
        class="modalContentWrapper"
        tabindex="-1"
      >
        <div>
          <div
            class="modalHeader"
          >
            <h3>
              Modal title
            </h3>
            <button
              class="MuiButton-root baseButton secondary"
              type="button"
            >
              x
            </button>
          </div>
          Some content
        </div>
        <div
          class="modalFooter"
        >
          <button
            class="MuiButton-root baseButton secondary"
            type="button"
          >
            Cancel
          </button>
          <button
            class="MuiButton-root baseButton error"
            type="button"
          >
            Confirm
          </button>
        </div>
      </div>
      <div
        data-testid="sentinelEnd"
        tabindex="0"
      />
    </div>
  </body>,
  "container": <div
    aria-hidden="true"
  >
    <div
      class="backdrop"
    />
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
